[Internal review] playwright: Update to latest versions and add new page objects

ndoschek opened the pull request:
- Update `playwright` to the latest version
- Extend page objects for compact and nested folders
- Extend ExplorerView tests with compact folders

<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

planger commented on examples/playwright/src/tests/theia-explorer-view.test.ts:
```diff
@@ -112,6 +112,8 @@ test.describe('Theia Explorer View', () => {
         // default setting `explorer.compactFolders=true` renders folders in a compact form - single child folders will be compressed in a combined tree element
         expect(await explorer.existsDirectoryNode('sampleFolderCompact/nestedFolder1/nestedFolder2', true /* compact */)).toBe(true);
+        // the `existsDirectoryNode` function will expand the folder, hence we wait for the file nodes to increase as we expect a txt child file node
+        await explorer.waitForFileNodesToIncrease(9);
```
I'd find it more reliable and clear, if we didn't put constant values as a parameter to this check, but rather query the number of file nodes before with
```typescript
const fileStatElements = await explorer.visibleFileStatNodes();
```
WDYT?

planger commented on examples/playwright/src/tests/theia-explorer-view.test.ts:
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
+    test('should delete nested folder "sampleDirectoryCompact/nestedFolder1/nestedFolder2"', async () => {
+        expect(await explorer.existsDirectoryNode('sampleDirectoryCompact/nestedFolder1/nestedFolder2', true /* compact */)).toBe(true);
+        await explorer.deleteNode('sampleDirectoryCompact/nestedFolder1/nestedFolder2', true /* confirm */, 'nestedFolder2' /* nodeSegmentLabel */);
+        await explorer.waitForFileNodesToDecrease(10);
```
Please see my comment above (avoiding constant numbers here).

planger commented on examples/playwright/src/tests/theia-explorer-view.test.ts:
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
+        await explorer.deleteNode('sampleDirectoryCompact/nestedFolder1/nestedFolder2', true /* confirm */, 'nestedFolder2' /* nodeSegmentLabel */);
+        await explorer.waitForFileNodesToDecrease(10);
+        const fileStatElements = await explorer.visibleFileStatNodes();
+        expect(fileStatElements.length).toBe(9);
```
Please see my comment above (avoiding constant numbers here).

planger commented on examples/playwright/src/tests/theia-explorer-view.test.ts:
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
+    test('should delete compact folder "sampleDirectoryCompact/nestedFolder1"', async () => {
+        expect(await explorer.existsDirectoryNode('sampleDirectoryCompact/nestedFolder1', true /* compact */)).toBe(true);
+        await explorer.deleteNode('sampleDirectoryCompact/nestedFolder1', true /* confirm */, 'sampleDirectoryCompact' /* nodeSegmentLabel */);
+        await explorer.waitForFileNodesToDecrease(9);
```
Please see my comment above (avoiding constant numbers here).

planger commented on examples/playwright/src/tests/theia-explorer-view.test.ts:
```diff
@@ -141,4 +143,44 @@ test.describe('Theia Explorer View', () => {
+        await explorer.deleteNode('sampleDirectoryCompact/nestedFolder1', true /* confirm */, 'sampleDirectoryCompact' /* nodeSegmentLabel */);
+        await explorer.waitForFileNodesToDecrease(9);
+        const fileStatElements = await explorer.visibleFileStatNodes();
+        expect(fileStatElements.length).toBe(8);
```
Please see my comment above (avoiding constant numbers here).

planger reviewed (approved):
Great, thank you! I just have one suggestion inline. Other than that, this looks perfect. Thank you!

ndoschek commented:
Thank you @planger for the fast review! I addressed your comments and opened the PR directly against Theia.
Closing this now in favor of https://github.com/eclipse-theia/theia/pull/12843
